### PR TITLE
ROC-2760: Add core case data store client

### PR DESCRIPTION
**Pull request by sabahirfan**
**JIRA link:**
https://tools.hmcts.net/jira/browse/ROC-2760

**Change description**
This PR adds client to save claim in core case data (CCD).

### CODE REVIEW

File: src/main/java/uk/gov/hmcts/cmc/ccd/client/StartCaseworkerApi.java [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+@FeignClient(name = "core-case-data-api", url = "${core_case_data.api.url}")
+public interface StartCaseworkerApi {
+
+    @RequestMapping(
+        method = RequestMethod.GET,
+        value = "/caseworkers/{userId}/jurisdictions/{jurisdictionId}/case-types/{caseType}/event-triggers/"
+            + "{eventId}/token",
+        headers = CONTENT_TYPE + "=" + APPLICATION_JSON_VALUE
+    )
+    ResponseEntity<StartEventResponse> start(@RequestHeader(AUTHORIZATION) String authorisation,
```
Comment (timja-kainos) [inline src/main/java/uk/gov/hmcts/cmc/ccd/client/StartCaseworkerApi.java]: `startForCaseWorker` perhaps?

File: src/main/java/uk/gov/hmcts/cmc/ccd/client/SubmitCaseworkerApi.java [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+import static org.springframework.http.HttpHeaders.AUTHORIZATION;
+
+@FeignClient(name = "core-case-data-api", url = "${core_case_data.api.url}")
+public interface SubmitCaseworkerApi {
+
+    @RequestMapping(
+        method = RequestMethod.POST,
+        value = "/caseworkers/{userId}/jurisdictions/{jurisdictionId}/case-types/{caseType}/cases"
+    )
+    ResponseEntity<CaseDetails> submit(@RequestHeader(AUTHORIZATION) String authorisation,
```
Comment (timja-kainos) [inline src/main/java/uk/gov/hmcts/cmc/ccd/client/SubmitCaseworkerApi.java]: same here

File: src/main/java/uk/gov/hmcts/cmc/ccd/client/model/CaseDetails.java [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+    private LocalDateTime createdDate;
+    @JsonProperty("last_modified")
+    private LocalDateTime lastModified;
+    private String state;
+    @JsonProperty("locked_by_user_id")
+    private Integer lockedBy;
+    @JsonProperty("security_level")
+    private Integer securityLevel;
+    @JsonProperty("case_data")
+    private Map<String, JsonNode> data;
```
Comment (timja-kainos) [inline src/main/java/uk/gov/hmcts/cmc/ccd/client/model/CaseDetails.java]: @rlewan thoughts?

Comment (rlewan) [reply src/main/java/uk/gov/hmcts/cmc/ccd/client/model/CaseDetails.java]: I think we shouldn't use a type that is transfer format related in our business types. It's mixing different concerns.

Comment (sabahirfan) [reply src/main/java/uk/gov/hmcts/cmc/ccd/client/model/CaseDetails.java]: done

File: src/main/java/uk/gov/hmcts/cmc/ccd/client/SubmitForCaseworkerApi.java [hunk 4]
```diff
@@ -0,0 +1,32 @@
...
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RequestMethod;
+import org.springframework.web.bind.annotation.RequestParam;
+import uk.gov.hmcts.cmc.ccd.client.model.CaseDataContent;
+import uk.gov.hmcts.cmc.ccd.client.model.CaseDetails;
+
+import static org.springframework.http.HttpHeaders.AUTHORIZATION;
+
+@FeignClient(name = "core-case-data-api", url = "${core_case_data.api.url}")
+public interface SubmitForCaseworkerApi {
```
Comment (timja-kainos) [inline src/main/java/uk/gov/hmcts/cmc/ccd/client/SubmitForCaseworkerApi.java]: not sure if we want multiple classes for caseworker and citizen,
are the calls quite different?

thoughts?

Comment (sabahirfan) [reply src/main/java/uk/gov/hmcts/cmc/ccd/client/SubmitForCaseworkerApi.java]: They have different endpoints for both. Each require different role etc. Not very sure about the other differences.

Comment (rlewan) [reply src/main/java/uk/gov/hmcts/cmc/ccd/client/SubmitForCaseworkerApi.java]: I suppose having them separate does not hurt too much? I guess with Feign we cannot share anything anyway, it's just call definitions?

Comment (sabahirfan) [reply src/main/java/uk/gov/hmcts/cmc/ccd/client/SubmitForCaseworkerApi.java]: May be combine both `start` and `submit` in one interface definition for `caseworker`.

Comment (dharmendrak) [reply src/main/java/uk/gov/hmcts/cmc/ccd/client/SubmitForCaseworkerApi.java]: we could you same client for both citizen and caseworkers, they can be put in path param

Comment (sabahirfan) [reply src/main/java/uk/gov/hmcts/cmc/ccd/client/SubmitForCaseworkerApi.java]: But that would be our assumption that might change later at CCD side.

Comment (timja-kainos) [reply src/main/java/uk/gov/hmcts/cmc/ccd/client/SubmitForCaseworkerApi.java]: I'm happy with it how it is

**Pull request merged**